packages/expo-file-system/android/src/main/java/expo/modules/filesystem/FileSystemModule.java

```diff
@@ -520,6 +523,35 @@ public void onResponse(Call call, Response response) throws IOException {
     }
   }
 
+  @ExpoMethod
+  public void getTotalDiskCapacityAsync(Promise promise) {
+    try {
+      StatFs root = new StatFs(Environment.getDataDirectory().getAbsolutePath());
+      long blockCount = root.getBlockCountLong();
+      long blockSize = root.getBlockSizeLong();
+      BigInteger capacity = BigInteger.valueOf(blockCount).multiply(BigInteger.valueOf(blockSize));
+      promise.resolve(capacity.doubleValue());
```

Comment:
How big could BigInteger be? Can it overflow?

Reply:
I cast it down to `Number.MAX_SAFE_INTEGER` if it exceeds the max integer JS could get.